apps/web/vibes/soul/sections/account-settings-section/update-account-form.tsx

```diff
@@ -67,17 +148,15 @@ export function UpdateAccountForm({
         key={fields.email.id}
         label={emailLabel}
       />
-      <SubmitButton>{submitLabel}</SubmitButton>
+      <Input
+        {...getInputProps(fields.company, { type: 'text' })}
+        errors={fields.company.errors}
+        key={fields.company.id}
+        label={companyLabel}
+      />
+      <Button name="intent" size="small" type="submit" value="update" variant="secondary">
+        {submitLabel}
+      </Button>
     </form>
   );
 }
-
-function SubmitButton({ children }: { children: React.ReactNode }) {
-  const { pending } = useFormStatus();
-
-  return (
-    <Button loading={pending} size="small" type="submit" variant="secondary">
-      {children}
-    </Button>
-  );
-}
```

Comment:
I'm curious if I should add this back in to get the form status state + button loading?

Reply:
I would say yes.

Reply:
Let's tackle this separately as I think we need to rethink how we surface the hooks for alerting. I don't have a good idea for it yet, but if y'all have suggestions let me know.

Reply:
For alerting, we do this right now:

```ts
useEffect(() => {
  if (form.status === 'success') toaster.success('Succeeded!')
}, [form.status])
```

Reply:
Ah... so that easier than I thought. For the localized strings, do we just pass it in as a prop to the component?

Reply:
For localization, do this:

```ts
const [{ lastResult, successMessage }, formAction] = useActionState(action, { lastResult: null })

const [form] = useForm({ lastResult })

useEffect(() => {
  if (lastResult.status === 'success') toaster.success(lastResult.successMessage)
}, [lastResult])
```

Reply:
We originally were passing it as a prop but we then realized that sometimes the localized message depends on the parameters. For example, in the PLP the empty state title uses the search term. We couldn't just pass it as a prop because the header is rendered in `layout.tsx`, which doesn't have access to `searchParams`. The solution was to have the message be part of the action state/payload. And this makes sense since the success state (or an error or empty state) are the outcome of the action